### PR TITLE
chore(tests): prune tautological and library-only tests

### PR DESCRIPTION
**Pull request by louistrue**
Focused cleanup of low-value TS tests flagged by a test-suite audit. Every flagged file was read and the audit claim verified before touching it; several flagged tests turned out to guard real invariants and were kept (listed below).

### apps/viewer Zustand slice tests (pure set-state-then-read-back and initial-state-defaults cases only)
- `selectionSlice.test.ts`: 34 → 27 cases. Deleted the 3 initial-state cases, 2 plain `setSelectedEntity` set/clear cases, and the 2 plain legacy `setSelectedEntityId` set/clear cases. **Kept** the "setSelectedEntity does NOT update selectedEntityId" case — it pins the two-selection-channels contract (global-id set drives renderer highlight; model-aware ref is properties-only).
- `annotationsSlice.test.ts`: 9 → 7 cases. Deleted the initial-state case and the plain "beginDraft opens a draft" set/read case. Kept commit logic, the empty-note drop branch, pin-deselect-on-new-draft, and localStorage persistence.
- `dataSlice.test.ts`: 22 → 15 cases. Deleted the 4 initial-state cases and the 3 identity-assignment `setIfcDataStore`/`setGeometryResult` cases. Kept all append/clone/fresh-copy/coordinate-info logic tests.
- `visibilitySlice.test.ts`: 36 → 29 cases. Deleted 2 empty-map initial-state cases, 1 plain hide case subsumed by the set-creation/model-separation cases, and the plain legacy `hideEntity` case (covered by show/toggle/isEntityVisible tests). Collapsed the 4 copy-pasted `toggleTypeVisibility` cases into one loop over all 5 type keys (adds `ifcGrid`, previously untested in isolation). **Kept** the persisted-preferences initial-state case (non-obvious localStorage-backed behavior).
- `pinboardSlice.test.ts`: **unchanged (0 deleted)**. Every case asserts cross-slice isolation sync, view-id branching, empty-input branches, or section-plane capture — none are pure set/read.
- `sectionSlice` / `scheduleSlice`: untouched, per scope.

### packages/lists `engine.test.ts`: 27 → 19 it-blocks (25 runtime cases)
Building a real data store in-package is impractical without new heavy deps: the package depends only on `@ifc-lite/data`/`@ifc-lite/encoding` (no parser), and the real `ListDataProvider` adapter lives in `apps/viewer/src/lib/lists/adapter.ts`. The mock is the engine's own provider port, and the cases do assert engine behavior (filtering, snapshots, grouping, CSV escaping, IFC-boolean resolution), so the suite was reduced rather than replaced:
- Collapsed the 5 per-source condition-filter cases + the empty-`entityTypes` targeting case into one `it.each` table (7 rows covering attribute/material/classification-code/classification-name/exists/spatial).
- Merged "snapshot drops foreign ids" + "snapshot honours conditions" into one case; kept the cross-model-bleed case separate.
- Merged the two `discoverColumns` aggregation cases (multi-provider, multi-type) into one.
- Replaced the two `LIST_PRESETS` tautologies ("at least 3 presets", fields-truthy loop) with one case that additionally **executes every preset** against the provider.

### packages/cli `mutate.test.ts`
- Collapsed the 8 near-identical `parseWhereFilter` operator cases into one `it.each` table (all 8 operators still covered). The remaining edge cases (throws, underscore pset names, empty value) kept as individual tests. (The audit counted 9; the 9th candidate, "handles pset names with underscores", is a distinct name-parsing edge, not an operator case — kept.)

### packages/extensions byte determinism
- `flavor/packer.test.ts`: replaced the hex-string byte-equality determinism check with a semantic one — two packs of the same logical content (extension-bundle Maps built in different insertion orders) must unpack to deep-equal payloads. Nothing in the codebase hashes or pins the packed `.iflv` envelope bytes; compatibility is via the `format`/`version` fields, so byte layout is not a contract there.
- `bundle/iflx.test.ts`: **KEPT the byte-for-byte determinism test** and documented why in the test: `.iflx` packed bytes ARE an on-disk integrity contract — `InstalledExtensionRecord.bundleHash` (and the flavor schema's per-extension `bundleHash`) is SHA-256 over the packed envelope, verified fail-closed on every load in `host/loader.ts`. (The Ed25519 signature covers the canonical file-map serialization, not the envelope, but the bundleHash check makes byte stability load-bearing regardless.)

## Kept — audit claims did not hold (packages/collab)
- `test/csg.test.ts`: not generic computational geometry — it tests our `src/geometry/csg.ts` CRDT helpers (`appendCSGOp`/`removeCSGOp`/`moveCSGOp` over the collab doc) including a cross-peer convergence case. KEPT.
- `test/color.test.ts`: not library behavior — `colorForUser`/`fnv1a` are first-party (`src/awareness/color.ts`) and used by the collab examples; cross-peer color determinism is the contract. KEPT.
- `test/parametric.test.ts`: does not test Yjs semantics at all — it tests our parametric mesh kernel (`src/geometry/parametric.ts`) and the mesh-hash determinism harness. KEPT.
- `test/perf-benchmark.test.ts`: has real assertions, not just timing output — it pins the spec §15 budgets (single-attr update ≤ 200 bytes, 1k-entity cold load ≤ 3 s, all benchmark `ok` flags). KEPT.
- `test/migration-sample.test.ts`: covers `src/doc/migration-ifc4-to-ifc4x3.ts`, which still exists and is registered via `installIfc4ToIfc4x3Migration` — this is the schema-version migration path, unrelated to the removed legacy STEP collab transport. KEPT.

## Verification
`pnpm turbo test --filter=@ifc-lite/lists --filter=@ifc-lite/collab --filter=@ifc-lite/extensions --filter=@ifc-lite/cli --filter=@ifc-lite/viewer` — all green:
- @ifc-lite/viewer: 939 passed, 2 skipped, 0 failed (node:test via tsx)
- @ifc-lite/extensions: 586 passed (52 files)
- @ifc-lite/collab: 136 passed (35 files)
- @ifc-lite/cli: 109 passed
- @ifc-lite/lists: 25 passed
`tsc --noEmit` clean for lists, cli, extensions.

Net: −342 lines of test code, +83.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Reorganized and reshuffled multiple test suites for clearer structure
  * Removed some redundant initial-state assertions to streamline tests
  * Enhanced parameterized tests to cover many operators and condition sources
  * Strengthened determinism checks for packaging and unpacking behaviors
  * Expanded list execution and preset validation scenarios to ensure stable outputs

### CODE REVIEW

File: packages/extensions/src/flavor/packer.test.ts [hunk 1]
```diff
@@ -48,10 +48,23 @@ describe('packFlavor / unpackFlavor', () => {
...
+    const ab = new Map([['com.example.a@1.0.0', bytesA], ['com.example.b@2.0.0', bytesB]]);
+    const ba = new Map([['com.example.b@2.0.0', bytesB], ['com.example.a@1.0.0', bytesA]]);
+
+    const r1 = unpackFlavor(packFlavor(flavor(), { extensionBundles: ab }));
+    const r2 = unpackFlavor(packFlavor(flavor(), { extensionBundles: ba }));
+    if (!r1.ok || !r2.ok) throw new Error('expected both packs to unpack');
+    expect(r2.value).toEqual(r1.value);
+    expect(Array.from(r1.value.extensionBundles.keys())).toEqual(
+      Array.from(r2.value.extensionBundles.keys()),
+    );
```
[the remark this answers is not in the document]
Comment (louistrue) [reply packages/extensions/src/flavor/packer.test.ts:83]: Addressed in 932feb68 — restored a byte-equality determinism test: packing the same flavor twice and packing a key-reordered bundle Map both assert identical bytes, so regressions in key sorting or gzip mtime pinning are caught again. Kept the semantic unpack-equality test alongside it.

File: apps/viewer/src/store/slices/visibilitySlice.test.ts [hunk 2]
```diff
@@ -286,28 +263,13 @@ describe('VisibilitySlice', () => {
...
-      const initial = state.typeVisibility.ifcAnnotations;
-      state.toggleTypeVisibility('ifcAnnotations');
-      assert.strictEqual(state.typeVisibility.ifcAnnotations, !initial);
+    it('should toggle each type key independently', () => {
+      const keys = ['spaces', 'openings', 'site', 'ifcAnnotations', 'ifcGrid'] as const;
+      for (const key of keys) {
+        const initial = state.typeVisibility[key];
+        state.toggleTypeVisibility(key);
+        assert.strictEqual(state.typeVisibility[key], !initial, `toggle ${key}`);
+      }
```
[the remark this answers is not in the document]
Comment (louistrue) [reply apps/viewer/src/store/slices/visibilitySlice.test.ts:280]: Addressed in 932feb68 — the loop now snapshots `{...state.typeVisibility}` before each toggle and asserts every other key is unchanged after toggling, so independence is actually validated.
